scripts/generateAssetData/generateRelatedAssetIndex/generateRelatedAssetIndex.ts

```diff
@@ -166,13 +167,27 @@ let sadCount = 0
 
 const processRelatedAssetIds = async (
   assetId: AssetId,
-  assetData: AssetsById,
+  assetData: Record<AssetId, PartialFields<Asset, 'relatedAssetKey'>>,
   relatedAssetIndex: Record<AssetId, AssetId[]>,
+  throttle: () => Promise<void>,
 ): Promise<void> => {
-  // don't fetch if we've already got the data from a previous request
   const existingRelatedAssetKey = assetData[assetId].relatedAssetKey
-  if (existingRelatedAssetKey) {
-    return
+  // We already have an existing relatedAssetKey, so we don't need to fetch it again
+  if (existingRelatedAssetKey !== undefined) return
```

Comment:
The second crux of this PR - as mentioned in https://github.com/shapeshift/web/pull/7408/files#r1688698060, `relatedAssetKey` will now *always* be included at generation time, effectively always `undefined` for now, and then set to `AssetId | undefined` by the generation script.

The next time it runs, if it encounters any `relatedAssetKey` which is not undefined, it means that we already know this asset and have successfully its related assets, so we can skip refetching.